sp/src/game/server/ez2/npc_basepredator.cpp

```diff
@@ -687,7 +687,23 @@ bool CNPC_BasePredator::CanMateWithTarget( CNPC_BasePredator * pTarget, bool rec
 //=========================================================
 bool CNPC_BasePredator::ShouldEatInCombat()
 {
-	return m_iMaxHealth > m_iHealth && HasCondition( COND_PREDATOR_SMELL_FOOD ) && !IsSameSpecies( GetEnemy() ) && ( !IsInSquad() || OccupyStrategySlot( SQUAD_SLOT_FEED ) );
+	// Do we need health?
+	if ( m_iHealth >= ( m_iMaxHealth * GetEatInCombatPercentHealth() ) )
```

Comment:
Is this confirmed to be working? It seems like the `GetEatInCombatPercentHealth` here would be automatically casted to an integer before the multiplication.

If you want to protect against that, you would cast `m_iMaxHealth` to a float first:
```suggestion
	if ( m_iHealth >= ( (float)m_iMaxHealth * GetEatInCombatPercentHealth() ) )
```
This isn't necessary if the comparison is already working as-is.

Reply:
Okay, we're quizzing my C++ knowledge here - GetEatInCombatPercentHealth() already returns a float. The result of an integer multiplied by a float should be a float, right? I don't believe that a cast is necessary here. I could be wrong though - please correct me if I am.

Reply:
I don't have a solid answer to that, which is why I'm asking if the comparison is already working as-is. Did you not test it to see if it functions properly?

Reply:
Yes, I tested it. This is what I was talking about with testing being difficult yesterday - it is still very possible my test was wrong as it's very hard to isolate this specific functionality from the rest of what the NPC does.

Reply:
Basic research online suggests that this comparison would indeed promote `m_iMaxHealth` to a float automatically before multiplying. I know for certain that I've needed to explicitly cast integers to floats before, but I guess that was only necessary in more specific cases? 🤷

For now, I'll take your word for it and assume this is working properly.